pandas/_libs/internals.pyx

```diff
@@ -370,22 +378,17 @@ def get_blkno_indexers(int64_t[:] blknos, bint group=True):
     # There's blkno in this function's name because it's used in block &
     # blockno handling.
     cdef:
-        int64_t cur_blkno
-        Py_ssize_t i, start, stop, n, diff
-
+        Py_ssize_t i, diff, tot_len, stop
+        Py_ssize_t start = 0, n = blknos.shape[0]
         object blkno
         object group_dict = defaultdict(list)
         int64_t[:] res_view
-
-    n = blknos.shape[0]
+        int64_t cur_blkno = blknos[start]
 
     if n == 0:
         return
 
-    start = 0
-    cur_blkno = blknos[start]
```

Comment:
doing this here instead of above means we can avoid the assignment in the n==0 case